server/src/main/java/io/crate/types/StringType.java

```diff
@@ -142,6 +155,24 @@ public String explicitCast(Object value) throws IllegalArgumentException, ClassC
         }
     }
 
+    @Override
+    public String valueForInsert(Object value) {
+        if (value == null) {
+            return null;
+        }
+        var string = convert(value);
```

Comment:
I think we should be able to assert that the value is a String. Or even change the signature (`valueForInsert(T value)`) and not run the implicit cast logic.

If the value didn't match the target type, we would already have added an implicit cast, so the value at this point must already match the type.

Reply:
I've added the assertion. Regarding the `valueForInsert(T)`, I am not sure how to do that, see https://github.com/crate/crate/pull/10033/files#diff-9cad83e8c7fa6eff3e32916ff493f709R75 it won't be probably possible to resolve the generic type, e.g. see https://github.com/crate/crate/pull/10033/files#diff-3bb55e4656a2a8f9d19f282021eff5f6R106 here the input argument has Object type.

Reply:
> I am not sure how to do that, it won't be probably possible to resolve the generic type

Ah right, let's leave it as it is now then